### DISCUSSION TITLE
How many types of transformations are there to apply to this sample stroke?

### DISCUSSION THREAD

**Discussion by nissansz** [python-pillow/Pillow#7410] [Q&A] [answered]
How many types of transformations are there to apply to this sample stroke?
I want to get a function to show all transformation effects, then random pick the desired transformations.

![tmpagkeeh30](https://github.com/python-pillow/Pillow/assets/41010669/c7c2a4d8-af67-4207-9016-1751c415a43b)

**Comment by radarhere** [comment, marked as answer]
When you say 'apply to this sample stroke', it's unclear if you mean that you want just the yellow pixels transformed, or if the whole image should be transformed.

Pillow is an image processing library. Depending on what your definition of 'transformation' is, asking what transformations can be applied is similar to asking for a list of everything that Pillow can do.

Are you aware of the documentation? https://pillow.readthedocs.io/en/stable/. You can read through the possible things to do with Pillow there.

**Comment by nissansz** [comment]
Thank you. I mean whole stroke image. 
I am trying to find a method to imitate handwritten stroke.
The original stroke is mainly curve from svg.
The effect is not very similar to handwritten when adjusted point values,
![image](https://github.com/python-pillow/Pillow/assets/41010669/90202e1f-2f67-45ca-97ea-0e89913602f2)

Real handwritten effect as below
![image](https://github.com/python-pillow/Pillow/assets/41010669/64ff42c7-6bc2-476b-94ca-c743ce67d506)

**Reply by radarhere** [reply]
You've asked about drawing handwritten text in #7379 and #7402. I really think the solution is to find a font that looks "handwritten".

**Reply by nissansz** [reply]
I saw some input methods, which can imitate handwritting styles with 10 sample characters using AI.
I wonder how to use this AI method in pillow draw.

**Reply by radarhere** [reply]
If you have to provide 10 characters as input, that doesn't sound simple to recreate. I don't see what the advantage would be over a handwritten font.

**Reply by nissansz** [reply]
Each person has different handwriting style, any good method to recognize or convert handwritten characters to printed characters?

**Reply by radarhere** [reply]
As [has been said](https://github.com/python-pillow/Pillow/issues/7376#issuecomment-1706317192), Pillow doesn't have any functionality for text detection.

**Reply by nissansz** [reply]
I mean other methods/codes that work fine.

**Reply by radarhere** [reply]
Code other than Pillow? Googling 'ocr handwriting python', I found https://blog.davidbramsay.com/handwriting-ocr-in-python/amp which lays out some alternatives.

**Reply by nissansz** [reply]
Thank you.
